Added Release Notes for Hazelcast Operator 5.6 (#73)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/release-notes.adoc

= 5.6 Release Notes

This release introduced the xref:high-availability-mode.adoc[High Availability Mode] configuration and custom resources (CRs) for xref:cache-configuration.adoc[Cache] and xref:queue-configuration.adoc[Queue] data structures. We also added persistence support for the Cache CR and provided Operator installation using Helm.

== New Features

* Introduced High Availability Mode configuration to create clusters that are resilient to node and zone failures (https://github.com/hazelcast/hazelcast-platform-operator/pull/538[#538])
* Added support for the Queue CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/428[#428])
* Added support for the Cache CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/430[#430])
* Users can enable persistence support for the Cache CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/474[#474])
* Replaced the `autoForceStart` option with the `startupAction` that allows the user to trigger Force Start or Partial Start when applied (https://github.com/hazelcast/hazelcast-platform-operator/pull/511[#511])
* Introduced Helm chart for Platform Operator installation

== Changes

> * Added connectivity check to WAN Replication CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/541[#541])

Reviewer: ```suggestion
* Added a connectivity check to the WAN Replication CR. When creating a WAN Replication CR, you can now check that the target cluster is reachable (https://github.com/hazelcast/hazelcast-platform-operator/pull/541[#541])
```
Again, all on one line?

> ** When creating a WAN Replication CR, a user will be sure that the target cluster is reachable

Reviewer: ```suggestion

```

> * Persist Wan Replication configuration into generated Hazelcast ConfigMap (https://github.com/hazelcast/hazelcast-platform-operator/pull/513[#513])

Reviewer: ```suggestion
* Added persistence for WAN Replication configurations. WAN Replication configurations are now stored in the Hazelcast ConfigMap so that when member containers restart, replication continues (https://github.com/hazelcast/hazelcast-platform-operator/pull/513[#513])
```

> ** Wan Replication configurations will be stored in Hazelcast ConfigMap. When member containers restart, they will be able to continue the replication

Reviewer: ```suggestion
```

> * Client recreation logic for times the client cannot access the cluster (https://github.com/hazelcast/hazelcast-platform-operator/pull/500[#500])

Reviewer: ```suggestion
* Added logic for recreating a Hazelcast client when the client is unable to access a cluster. When the client is unable to connect, the Hazelcast Platform Operator restarts it (https://github.com/hazelcast/hazelcast-platform-operator/pull/500[#500])
```

> ** The Platform Operator will restart the Hazelcast client when the cluster is inaccessible

Reviewer: ```suggestion
```

> * Converted existing CR validation to validating webhook (https://github.com/hazelcast/hazelcast-platform-operator/pull/492[#492])

Reviewer: ```suggestion
* Updated CR validation to use a webhook. The Hazelcast Platform Operator can now report issues with CRs before they are applied (https://github.com/hazelcast/hazelcast-platform-operator/pull/492[#492])
```

> ** This change allows the Operator to report issues with CRs before they are applied

Reviewer: ```suggestion
```

> * Unified the approach for exposing API fields (https://github.com/hazelcast/hazelcast-platform-operator/pull/498[#498])

Reviewer: ```suggestion
* Improved the approach for exposing API fields. Exposed API fields are now more consistent  (https://github.com/hazelcast/hazelcast-platform-operator/pull/498[#498])
```

> ** Exposed API fields are now more consistent with each other

Reviewer: ```suggestion
```

> * Removed `h.Spec.Persistence.DataLoadTimeoutSec` parameter (https://github.com/hazelcast/hazelcast-platform-operator/pull/512[#512])

Reviewer: ```suggestion
* Removed the `h.Spec.Persistence.DataLoadTimeoutSec` parameter (https://github.com/hazelcast/hazelcast-platform-operator/pull/512[#512])
```

* Split `restore` and `restore_local` commands (https://github.com/hazelcast/hazelcast-platform-operator/pull/471[#471])

> * Used Go client for cluster activation logic (https://github.com/hazelcast/hazelcast-platform-operator/pull/466[#466])

Reviewer: ```suggestion
* Updated the cluster activation logic to use the Go client (https://github.com/hazelcast/hazelcast-platform-operator/pull/466[#466])
```

> * Added tilt-debug support  (https://github.com/hazelcast/hazelcast-platform-operator/pull/528[#528])

Reviewer: ```suggestion
* Added `tilt-debug` support for the remote debugging of local Kubernetes clusters  (https://github.com/hazelcast/hazelcast-platform-operator/pull/528[#528])
```

* Added `ClusterIDs` to Phone Home data (https://github.com/hazelcast/hazelcast-platform-operator/pull/527[#527])

> * Changed capabilities at CSV from Seamless Upgrades to Full Lifecycle, which is become visible on OCP OperatorHub dashboard(https://github.com/hazelcast/hazelcast-platform-operator/pull/476[#476])

Reviewer: ```suggestion
* Updated capabilities of Cluster Service Version (CSV) from Seamless Upgrades to Full Lifecycle. Both options now appear on the OCP OperatorHub dashboard (https://github.com/hazelcast/hazelcast-platform-operator/pull/476[#476])
```

Author: @angelasimms In this case, both options do not appear on the dashboard. One option just replaced another one.

Reviewer: @cheels. I've slightly reworded. Does that work?

== Fixed Issues

> * Allowed creating HotBackup before the Hazelcast cluster is ready (https://github.com/hazelcast/hazelcast-platform-operator/pull/503[#503])

Reviewer: ```suggestion
* A HotBackup can now be created before the Hazelcast cluster is ready (https://github.com/hazelcast/hazelcast-platform-operator/pull/503[#503])
```

> * Added webhook support for OLM (https://github.com/hazelcast/hazelcast-platform-operator/pull/540[#540])

Reviewer: ```suggestion
* Added webhook support for Operator Lifecycle Management (OLM) (https://github.com/hazelcast/hazelcast-platform-operator/pull/540[#540])
```

> * Set Hazelcast resource message on non-existing hot backup (https://github.com/hazelcast/hazelcast-platform-operator/pull/514[#514])

Reviewer: ```suggestion
* Sets a Hazelcast resource message when attempting to restore a hot backup that does not exist (https://github.com/hazelcast/hazelcast-platform-operator/pull/514[#514])
```

> * Fixed Hazelcast ConfigMap when multiple CRs for Data Structures are created (https://github.com/hazelcast/hazelcast-platform-operator/pull/501[#501])

Reviewer: ```suggestion
* Multiple CRs for data structures can now be created and then stored correctly in the ConfigMap (https://github.com/hazelcast/hazelcast-platform-operator/pull/501[#501])
```

> ** The user can create multiple CR, and all will be stored in the ConfigMap

Reviewer: ```suggestion
```

> * Added security context to all containers for hostPath permissions (https://github.com/hazelcast/hazelcast-platform-operator/pull/485[#485])

Reviewer: ```suggestion
* Added a security context to all containers for hostPath permissions (https://github.com/hazelcast/hazelcast-platform-operator/pull/485[#485])
```

> * Fixed wrong variable assignment in WanRep ConfigMap (https://github.com/hazelcast/hazelcast-platform-operator/pull/554[#554])

Reviewer: ```suggestion
* Fixed the incorrect assignment of a variable in WanRep ConfigMap. After a restart, cluster members will now continue replicating map entries (https://github.com/hazelcast/hazelcast-platform-operator/pull/554[#554])
```

> ** After members restart, they will be able to continue replicating the map entries using Wan Replication

Reviewer: ```suggestion
```

> * Fixed empty pod name field in Hazelcast member status (https://github.com/hazelcast/hazelcast-platform-operator/pull/456[#456])

Reviewer: ```suggestion
* Hazelcast member status no longer contains an empty pod name field (https://github.com/hazelcast/hazelcast-platform-operator/pull/456[#456])
```
